src/libcharon/plugins/vici/vici_control.c

```diff
@@ -689,6 +689,35 @@ CALLBACK(reload_settings, vici_message_t*,
 	return send_reply(this, "reloading '%s' failed", lib->conf);
 }
 
+CALLBACK(activate_sa, vici_message_t*,
+	private_vici_control_t *this, char *name, u_int id, vici_message_t *request)
+{
+	int ike_id = request->get_int(request, -1, "ike-id");
+	if (-1 == ike_id)
+		return send_reply(this, "activating IKE_SA failed - missing 'ike-id'");
+
+	enumerator_t* isas = charon->controller->create_ike_sa_enumerator(
+		charon->controller, /* block */TRUE);
+
+	ike_sa_t *ike_sa;
+	bool found = FALSE;
+	while (isas->enumerate(isas, &ike_sa))
```

Comment:
Can replace this loop with `charon->ike_sa_manager->checkout_by_id()`. Don't forget to `checkin()` it afterwards.

Reply:
I will still have to go over the rest of `ike_sa`s to signal standby for all except the one to activate.
So what will be the benefit of `checkout_by_id()`?

Reply:
I was commenting on the code as it is now, which only looks for a specific IKE SA. If you have changed planned in the future then you can ignore my comment.